data/src/test/java/org/apache/iceberg/data/DeleteReadTests.java

```diff
@@ -330,6 +420,15 @@ private StructLikeSet rowSetWithoutIds(int... idsToRemove) {
     return set;
   }
 
+  private StructLikeSet rowSetWithoutIds(Set<Integer> idSet, Table iTable, List<Record> recordList) {
+    StructLikeSet set = StructLikeSet.create(iTable.schema().asStruct());
+    recordList.stream()
+        .filter(row -> !idSet.contains(row.getField("id")))
+        .map(record -> new InternalRecordWrapper(iTable.schema().asStruct()).wrap(record))
+        .forEach(set::add);
+    return set;
+  }
```

Comment:
I think the other `rowSetWithoutIds` method should be rewritten to call this one since it is more generic. That will also avoid the wrapper problem in the future.

Reply:
ok i'll rewrite the origin rowSetWithoutIds() method